spec/activerecord-bitemporal/bitemporal_scope_spec.rb

```diff
@@ -777,6 +777,93 @@ class Article < ActiveRecord::Base
       end
     end
 
+    describe "ActiveRecord::Bitemporal.transaction_at" do
+      let(:transaction_datetime) { "2019/05/05".in_time_zone }
+      let(:relation) { ActiveRecord::Bitemporal.transaction_at(transaction_datetime) { User.all } }
+      it { is_expected.to have_valid_at(time_current, table: "users") }
+      it { is_expected.to have_transaction_at(transaction_datetime, table: "users") }
+
+      context "transaction_datetime is nil" do
+        let(:transaction_datetime) { nil }
+        it { is_expected.to have_bitemporal_at(time_current, table: "users") }
+      end
+
+      describe ".transaction_at" do
+        let(:user_transaction_datetime) { "2019/05/05".in_time_zone }
+        let(:relation) { ActiveRecord::Bitemporal.transaction_at(transaction_datetime) { User.transaction_at(user_transaction_datetime) } }
+        it { is_expected.to have_valid_at(time_current, table: "users") }
+        it { is_expected.to have_transaction_at(user_transaction_datetime, table: "users") }
+      end
+
+      context "with ActiveRecord::Bitemporal.ignore_transaction_datetime" do
+        let(:relation) { ActiveRecord::Bitemporal.transaction_at(transaction_datetime) { ActiveRecord::Bitemporal.ignore_transaction_datetime { User.all } } }
+        it { is_expected.to have_valid_at(time_current, table: "users") }
+        it { is_expected.to not_have_transaction_at(table: "users") }
+      end
+
+      context ".ignore_transaction_datetime" do
+        let(:relation) { ActiveRecord::Bitemporal.transaction_at(transaction_datetime) { User.ignore_transaction_datetime } }
+        it { is_expected.to have_valid_at(time_current, table: "users") }
+        it { is_expected.to not_have_transaction_at(table: "users") }
+      end
+
+      describe "ActiveRecord::Bitemporal.transaction_at!" do
+        let(:transaction_datetime2) { "2019/06/06".in_time_zone }
+        let(:relation) { ActiveRecord::Bitemporal.transaction_at(transaction_datetime) { ActiveRecord::Bitemporal.transaction_at(transaction_datetime2) { User.all } } }
+        it { is_expected.to have_valid_at(time_current, table: "users") }
+        it { is_expected.to have_transaction_at(transaction_datetime2, table: "users") }
```

Comment:
Does this test case mean "Inner `ActiveRecord::Bitemporal.transaction_at` takes precedence in nested cases"? Why the test is nested in `describe "ActiveRecord::Bitemporal.transaction_at!"`?

Reply:
Thanks! Fix it.
https://github.com/kufu/activerecord-bitemporal/pull/82/commits/de492febf7ddaadc308d32c23ca9b19f54eaaccb